owncloud/owncloud.py

```diff
@@ -726,20 +726,39 @@ def user_exists(self, user_name):
         :raises: ResponseError in case an HTTP error status was returned
 
         """
+
+#
+# LDAP users can have a space in their names; ownCloud does not support
+# spaces in usernames so the space gets converted to an underscore.
+# When we search for the ldap user with a space in the name, we need
+# to remove the underscore to find the user.  To do this, if the username
+# being searched for has an underscore in it, we will search for users with
+# names that begin with the input username up to the underscore and then search
+# for the actual username in the result.  
+# If the name of the user we're looking for is ldap_user, we will search
+# for all users that have ldap in the name and then look explicitly for 
+# ldap_user in the results.
```

Comment:
Shouldn't we rather fix this in the server instead ?
If not in the users backend, then at least do this in the provisioning API module ?

CC @blizzz